protocols/mdns/src/behaviour/iface.rs

```diff
@@ -200,51 +200,58 @@ where
         cx: &mut Context,
         params: &impl PollParameters,
     ) -> Option<(PeerId, Multiaddr, Instant)> {
-        // Poll receive socket.
-        while let Poll::Ready(data) =
-            Pin::new(&mut self.recv_socket).poll_read(cx, &mut self.recv_buffer)
-        {
-            match data {
-                Ok((len, from)) => {
-                    if let Some(packet) = MdnsPacket::new_from_bytes(&self.recv_buffer[..len], from)
-                    {
-                        self.inject_mdns_packet(packet, params);
+        loop {
+            // 1st priority: Low latency: Create packet ASAP after timeout.
+            if Pin::new(&mut self.timeout).poll_next(cx).is_ready() {
+                log::trace!("sending query on iface {}", self.addr);
+                self.send_buffer.push_back(build_query());
+            }
+
+            // 2nd priority: Keep local buffers small: Empty buffers ASAP.
```

Comment:
Shouldn't we first try to send existing work (`self.send_buffer.pop_front`) and then add more work (`self.send_buffer.push_back`)? In case we do a `push_back`, we can do a `continue` to follow with a `pop_front`.

Reply:
I am not sure if it makes a big difference. The timer is our only local work, i.e. nothing else will push to the buffer. When we say "prefer local work over remote work", does that not imply that whatever generates the local work has the highest priority, followed by trying to send it to the remote?